catalog/imaging/catalog-video-transformer/src/test/java/com/connexta/transformer/video/TestMpegTsInputTransformer.java

```diff
@@ -0,0 +1,177 @@
+                        throw new Stanag4609ParseException(e);
+                    }
+                }
+
+        );
+
+        KlvProcessor klvProcessor = mock(KlvProcessor.class);
+
+        MpegTsInputTransformer t = new MpegTsInputTransformer(inputTransformer,
+                metacardTypes,
+                stanag4609Processor,
+                klvHandlerFactory,
+                defaultKlvHandler,
+                stanagParserFactory,
+                klvProcessor);
+
+        try (InputStream inputStream = new ByteArrayInputStream(new byte[] {})) {
+
+            Metacard finalMetacard = t.transform(inputStream);
+
+            assertThat(finalMetacard.getContentTypeName(), is(MpegTsInputTransformer.CONTENT_TYPE));
+            assertThat(finalMetacard.getMetadata(), is("the metadata"));
+
+        }
+
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testStanagParseError() throws Exception {
+        InputTransformer inputTransformer = mock(InputTransformer.class);
+        MetacardImpl metacard = new MetacardImpl(BasicTypes.BASIC_METACARD);
+        when(inputTransformer.transform(any(), any())).thenReturn(metacard);
+
+        List<MetacardType> metacardTypes = Collections.singletonList(mock(MetacardType.class));
+
+        Stanag4609Processor stanag4609Processor = mock(Stanag4609Processor.class);
+
+        KlvHandlerFactory klvHandlerFactory = mock(KlvHandlerFactory.class);
+
+        KlvHandler defaultKlvHandler = mock(KlvHandler.class);
+
+        StanagParserFactory stanagParserFactory = mock(StanagParserFactory.class);
+        Stanag4609TransportStreamParser streamParser = mock(Stanag4609TransportStreamParser.class);
+        when(streamParser.parse()).thenThrow(new RuntimeException());
+        when(stanagParserFactory.createParser(any())).thenReturn(() -> {
+            try {
+                return streamParser.parse();
+            } catch (Exception e) {
+                throw new Stanag4609ParseException(e);
+            }
+        });
+
+        KlvProcessor klvProcessor = mock(KlvProcessor.class);
+
+        MpegTsInputTransformer t = new MpegTsInputTransformer(inputTransformer,
+                metacardTypes,
+                stanag4609Processor,
+                klvHandlerFactory,
+                defaultKlvHandler,
+                stanagParserFactory,
+                klvProcessor);
+
+        try (InputStream inputStream = new ByteArrayInputStream(new byte[] {})) {
+            t.transform(inputStream);
+        }
+
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testInputStreamReadError() throws Exception {
+        InputTransformer inputTransformer = mock(InputTransformer.class);
+        MetacardImpl metacard = new MetacardImpl(BasicTypes.BASIC_METACARD);
+        when(inputTransformer.transform(any(), any())).thenReturn(metacard);
+
+        List<MetacardType> metacardTypes = Collections.singletonList(mock(MetacardType.class));
```

Comment:
fix, i went through the test code and pulled out as much common code as possible